packages/react-native-renderer/src/ReactNativeGetListeners.js

```diff
@@ -0,0 +1,168 @@
+  // Now, it is also possible to have N listeners
+  // for a specific event on a node. Thus, we accumulate all of the listeners,
+  // including the props listener, and return a function that calls them all in
+  // order, starting with the handler prop and then the listeners in order.
+  // We return either a non-empty array or null.
+  const listeners = [];
+  if (listener) {
+    listeners.push(listener);
+  }
+
+  // TODO: for now, all of these events get an `rn:` prefix to enforce
+  // that the user knows they're only getting non-W3C-compliant events
+  // through this imperative event API.
+  // Events might not necessarily be noncompliant, but we currently have
+  // no verification that /any/ events are compliant.
+  // Thus, we prefix to ensure no collision with W3C event names.
+  const requestedPhaseIsCapture = phase === 'captured';
+  const mangledImperativeRegistrationName = requestedPhaseIsCapture
+    ? 'rn:' + registrationName.replace(/Capture$/, '')
+    : 'rn:' + registrationName;
+
+  // Get imperative event listeners for this event
+  if (
+    stateNode.canonical._eventListeners[mangledImperativeRegistrationName] &&
+    stateNode.canonical._eventListeners[mangledImperativeRegistrationName]
+      .length > 0
+  ) {
+    const eventListeners =
+      stateNode.canonical._eventListeners[mangledImperativeRegistrationName];
+
+    eventListeners.forEach(listenerObj => {
+      // Make sure phase of listener matches requested phase
+      const isCaptureEvent =
+        listenerObj.options.capture != null && listenerObj.options.capture;
+      if (isCaptureEvent !== requestedPhaseIsCapture) {
+        return;
+      }
+
+      // For now (this is an area of future optimization) we must wrap
+      // all imperative event listeners in a function to unwrap the SyntheticEvent
+      // and pass them an Event.
+      // When this API is more stable and used more frequently, we can revisit.
+      const listenerFnWrapper = function(syntheticEvent) {
+        const eventInst = new CustomEvent(mangledImperativeRegistrationName, {
+          detail: syntheticEvent.nativeEvent,
+        });
+        eventInst.isTrusted = true;
+        // setSyntheticEvent is present on the React Native Event shim.
+        // It is used to forward method calls on Event to the underlying SyntheticEvent.
+        // $FlowFixMe
+        eventInst.setSyntheticEvent(syntheticEvent);
+
+        listenerObj.listener(eventInst);
+      };
+
+      // Only call once?
+      // If so, we ensure that it's only called once by setting a flag
+      // and by removing it from eventListeners once it is called (but only
+      // when it's actually been executed).
+      if (listenerObj.options.once) {
+        listeners.push(function(...args) {
+          // Remove from the event listener once it's been called
+          stateNode.canonical.removeEventListener_unstable(
+            mangledImperativeRegistrationName,
+            listenerFnWrapper,
```

Comment:
Also sorry I was trying to follow this but still confused, why do we pass listenerFnWrapper here? Should it not be listenerObj.listener?

Reply:
Hmm.  Yes! I believe you're right.